util/io/files/tar.go

```diff
@@ -83,7 +83,12 @@ func Untgz(dstPath string, r io.Reader, maxSize int64, preserveFileMode bool) er
 			continue
 		}
 
-		target := filepath.Join(dstPath, header.Name)
+		// Sanitize header.Name to prevent directory traversal attacks
+		cleanName := filepath.Clean(header.Name)
+		if strings.HasPrefix(cleanName, "..") || filepath.IsAbs(cleanName) {
```

Comment:
I am concerned there may be other inputs that can lead to traversal, but we would not catch them with our custom written logic - although the code looks good to me.

How about https://go.dev/blog/osroot#introducing-osroot, since we are on `go` that is new enough?